src/main/java/com/synopsys/integration/jenkins/scan/service/ScannerArgumentService.java

```diff
@@ -31,14 +42,26 @@ public List<String> getCommandLineArgs(Map<String, Object> scanParameters) {
 
         BlackDuckParametersService blackDuckParametersService = new BlackDuckParametersService();
         BlackDuck blackDuck = blackDuckParametersService.prepareBlackDuckInputForBridge(scanParameters);
-        commandLineArgs.add(createBlackDuckInputJson(blackDuck));
+
+        SCMRepositoryService scmRepositoryService = new SCMRepositoryService(listener, envVars);
+        Object scmObject =  scmRepositoryService.fetchSCMRepositoryDetails(scanParameters);
+
+        // added the Prcomment and Fixpr condition to create BlackDuck Input Json on the following code
+        if(scmObject instanceof Bitbucket) {
+            Bitbucket bitbucket = (Bitbucket) scmObject;
+            commandLineArgs.add(createBlackDuckInputJson( blackDuck, blackDuck.getAutomation().getPrcomment() || blackDuck.getAutomation().getFixpr() ? bitbucket : null));
+        }
```

Comment:
do we need the if condition here? 

we can directly use 

we are already checking the `instanceof` in `createBlackDuckInputJson` method

`commandLineArgs.add(createBlackDuckInputJson( blackDuck, blackDuck.getAutomation().getPrcomment() || blackDuck.getAutomation().getFixpr() ? scmObject : null));`

Reply:
We can have github/gitlab/bitbucket any type of scm, right? If we have to configure it only for bitbucket then the if condition is not needed. Otherwise for the bigger picture I think we may need to use the If condition.

Reply:
What I meant is we are already doing this here in  `createBlackDuckInputJson`method - 
```
if (scm instanceof Bitbucket) {
        Bitbucket bitbucket = (Bitbucket) scm;
        bridgeInput.setBitbucket(bitbucket);
 }
```
so we can directly pass the `scmObject` from here, we can do the future changes in  `createBlackDuckInputJson`method

Reply:
Yes. I got it.

Reply:
Also please remove the extra space at the starting of `blackDuck` parameter